src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -166,20 +164,28 @@ public override async Task<IEnumerable<string>> EnumerateVersionsAsync(PackageUR
                 string feedUrl = (purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT).EnsureTrailingSlash();
                 HttpClient httpClient = CreateHttpClient();
 
-                string? content = await GetHttpStringCache(httpClient, $"{feedUrl}{packageNamespace}/{packageName}/maven-metadata.xml", useCache);
+                string? content = await GetHttpStringCache(httpClient, $"{feedUrl}{packageNamespace}/{packageName}/", useCache);
                 List<string> versionList = new();
                 if (string.IsNullOrWhiteSpace(content))
                 {
                     return new List<string>();
                 }
-                XmlDocument doc = new();
-                doc.LoadXml(content);
-                foreach (XmlNode? versionObject in doc.GetElementsByTagName("version"))
+                
+                // Parse the html file.
+                HtmlParser parser = new();
+                AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(content);
+                
+                // Break the version content down into its individual lines. Includes the parent directory and xml + hash files.
+                IEnumerable<string> htmlEntries = document.QuerySelector("#contents").QuerySelectorAll("a").Select(a => a.TextContent);
+
+                foreach (string htmlEntry in htmlEntries)
                 {
-                    if (versionObject != null)
+                    // Get the version.
+                    if (htmlEntry.EndsWith('/') && !htmlEntry.Equals("../"))
                     {
-                        Logger.Debug("Identified {0} version {1}.", packageName, versionObject.InnerText);
-                        versionList.Add(versionObject.InnerText);
+                        var versionStr = htmlEntry.TrimEnd(htmlEntry[^1]);
+                        Logger.Debug("Identified {0} version {1}.", packageName, versionStr);
```

Comment:
Does this logger statement need to be here?

Reply:
I guess not? But it's there from the previous code, so I just kept it. We don't see it in our logs I don't think.